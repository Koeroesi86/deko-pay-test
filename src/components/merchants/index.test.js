import React from 'react';
import { shallow, mount } from 'enzyme';
import axios from 'axios';
import Merchants from './index';
import { act } from "@testing-library/react";

const mockMerchants = [
  {
    "merchant_id": "A1201",
    "name": "Rohan LLC Shoes",
    "_links": {
      "self": {
        "href": "/merchants/A1201.json"
      }
    }
  },
  {
    "merchant_id": "A1202",
    "name": "Carroll - Lang Grocery",
    "_links": {
      "self": {
        "href": "/merchants/A1202.json"
      }
    }
  }
];

jest.mock('axios');
axios.get.mockImplementation(() => Promise.resolve({ data: mockMerchants }));

describe('The Merchants component', () => {
  it('renders', () => {
    const result = shallow(<Merchants />);
    expect(result).toMatchSnapshot();
  });

  it('mounts', async () => {
    const setSelected = jest.fn();
    let result;
    await act(async () => {
      result = mount(<Merchants selected="A1201" setSelected={setSelected} />);
    });
    result.update();

    expect(axios.get).toHaveBeenCalledWith('http://interview.dekopay.com.s3.eu-west-2.amazonaws.com/merchants.json');
    expect(setSelected).toHaveBeenCalledWith('A1201');
    expect(result).toMatchSnapshot();
  });
});
