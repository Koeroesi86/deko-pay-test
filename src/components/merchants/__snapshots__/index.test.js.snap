// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Merchants component mounts 1`] = `
<div
  className="Merchants"
>
  <div
    className="header"
  >
    Merchants
  </div>
  <div
    className="merchant selected"
    key="merchant-A1201"
    onClick={[Function]}
  >
    <div
      className="name"
    >
      Rohan LLC Shoes
    </div>
    <svg
      className="arrow"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.17 32.92l9.17-9.17-9.17-9.17L20 11.75l12 12-12 12z"
      />
    </svg>
  </div>
  <div
    className="merchant "
    key="merchant-A1202"
    onClick={[Function]}
  >
    <div
      className="name"
    >
      Carroll - Lang Grocery
    </div>
    <svg
      className="arrow"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.17 32.92l9.17-9.17-9.17-9.17L20 11.75l12 12-12 12z"
      />
    </svg>
  </div>
</div>
`;

exports[`The Merchants component renders 1`] = `
<div
  className="Merchants"
>
  <div
    className="header"
  >
    Merchants
  </div>
</div>
`;
